### PR TITLE
[Improvement][Master] LowerWeightRoundRobin doSelect may be null

### PR DESCRIPTION
**Pull request by tracehh**
## Verify this pull request

This pull request is code cleanup without any test coverage.

*(or)*

This pull request is already covered by existing tests, such as *(please describe tests)*.

(or)

This change added tests and can be verified as follows:

(or)

If your pull request contain incompatible change, you should also add it to `docs/docs/en/guide/upgrede/incompatible.md`

### CODE REVIEW

**Review by ruanwenjun — COMMENTED** [review]
Can you add ut for this change?

**Comment by tracehh** [comment]
OK added UT @ruanwenjun

**Comment by ruanwenjun** [comment]
Please rebase the upstream-dev to fix the ci.

File: dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/dispatch/host/assign/LowerWeightRoundRobin.java [hunk 1]
```diff
@@ -58,14 +62,17 @@ private List<HostWeight> canAssignTaskHost(Collection<HostWeight> sources) {
         if (!zeroWaitingTask.isEmpty()) {
             return zeroWaitingTask;
         }
-        HostWeight hostWeight = sources.stream().min(Comparator.comparing(HostWeight::getWaitingTaskCount)).get();
-        List<HostWeight> waitingTask = Lists.newArrayList(hostWeight);
-        List<HostWeight> equalWaitingTask = sources.stream().filter(h -> !h.getHost().equals(hostWeight.getHost()) && h.getWaitingTaskCount() == hostWeight.getWaitingTaskCount())
-            .collect(Collectors.toList());
-        if (!equalWaitingTask.isEmpty()) {
-            waitingTask.addAll(equalWaitingTask);
+        if (!CollectionUtils.isEmpty(sources)) {
```
Comment (ruanwenjun) [inline dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/dispatch/host/assign/LowerWeightRoundRobin.java]: Why don't put this in begin.

File: dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/dispatch/host/assign/LowerWeightRoundRobin.java [hunk 2]
```diff
@@ -49,23 +51,28 @@ public HostWeight doSelect(Collection<HostWeight> sources) {
...
-        if (!zeroWaitingTask.isEmpty()) {
-            return zeroWaitingTask;
-        }
-        HostWeight hostWeight = sources.stream().min(Comparator.comparing(HostWeight::getWaitingTaskCount)).get();
-        List<HostWeight> waitingTask = Lists.newArrayList(hostWeight);
-        List<HostWeight> equalWaitingTask = sources.stream().filter(h -> !h.getHost().equals(hostWeight.getHost()) && h.getWaitingTaskCount() == hostWeight.getWaitingTaskCount())
-            .collect(Collectors.toList());
-        if (!equalWaitingTask.isEmpty()) {
-            waitingTask.addAll(equalWaitingTask);
+        if (!CollectionUtils.isEmpty(sources)) {
```
Comment (ruanwenjun) [inline dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/dispatch/host/assign/LowerWeightRoundRobin.java]:
```suggestion
        if (CollectionUtils.isEmpty(sources)) {
                return Collections.emptyList();
        }
```

Comment (tracehh) [reply dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/dispatch/host/assign/LowerWeightRoundRobin.java]: update it

Comment (tracehh) [reply dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/dispatch/host/assign/LowerWeightRoundRobin.java]: @ruanwenjun fixed

**Review by ruanwenjun — APPROVED** [review]
LGTM

**Pull request merged**